jupyter_rsession_proxy/__init__.py

```diff
@@ -29,13 +32,46 @@ def setup_rserver():
     def _get_env(port):
         return dict(USER=getpass.getuser())
 
+    def db_config():
+        '''Create a temporary directory to hold rserver's database.'''
+        db_dir = tempfile.TemporaryDirectory()
+        # create the rserver database config
+        db_conf = dedent("""
+            provider=sqlite
```

Comment:
What is it storing in this database? When does this temporary directory get cleaned up? On process exit? Also should the temp file be inside this directory?

Reply:
The [documentation](https://docs.rstudio.com/ide/server-pro/latest/database.html) doesn't say what is in the database, but [the source](https://github.com/rstudio/rstudio/tree/master/src/cpp/server/db) and an instance of the sqlite file show that the tables contains tables for revoked cookies. As of 1.4.x, rserver will fail if it is invoked as a non-privileged user and if you don't alter its default path of /var/lib/rstudio-server/rstudio.sqlite. 

I'm not too worried about cleaning up anything in a TMP_DIR. The name is randomized so there shouldn't be a conflict and as far as I can tell, the database won't get too large. The config file doesn't have to be in the temporary directory containing the database, but it could be. It'd be tidier that way I suppose.

Shiny server has similar assumptions about privileged locations which is why we [have to move them](https://github.com/ryanlovett/jupyter-shiny-proxy/blob/master/jupyter_shiny_proxy/__init__.py#L11). We're putting Shiny's data in the user's home dir, however the rserver docs say the database must not be on NFS which is why I chose a temporary location.

Reply:
Cool, that makes sense. In /tmp, it will go away on server stop. That is probably ok tho.